apps/account/views.py

```diff
@@ -42,6 +44,27 @@ def render_to_response(self, context, **response_kwargs):
         return response
 
 
+class AccountDeletionView(LoginRequiredMixin, SuccessMessageMixin, generic.DeleteView):
+    template_name = "a4_candy_account/account_deletion.html"
+    form_class = forms.AccountDeletionForm
+    success_message = _("Your account was successfully deleted.")
+    success_url = "/"
+
+    def get_object(self):
+        return get_object_or_404(User, pk=self.request.user.id)
+
+    def form_valid(self, form):
+        user = self.request.user
+        password = form.cleaned_data.get("password")
+        if not user.check_password(password):
+            form.add_error(
+                "password", ValidationError("Incorrect password.", "invalid")
+            )
+            return super().form_invalid(form)
+        logout(self.request)
+        return super().form_valid(form)
```

Comment:
@goapunk you can use a ModelForm. The field needs also the `label`, and `strip` set to False, as i see from django's AuthUser forms, it is set to [False](https://github.com/django/django/blob/main/django/contrib/auth/forms.py#L295) there too, so looks like it accepts spaces too.
For the user check @hom3mad3 @mcastro-lqd may be good to add an email field as well, though I know is not part of the design. Or add itbut the design was not there when we discussed the story.

Reply:
@m4ra will look into the `ModelForm`. I don't understand why we would need the email, it doesn't add any security?

Reply:
I've added the label and strip, however I struggle to make this work with the `ModelForm` as the user instance is only created after validation (i.e. when calling `save()`). If you have an idea how to solve this could you take over @m4ra ?

Reply:
OK, let's leave it as it is then.